api/src/services/opportunities_v1/search_opportunities.py

```diff
@@ -148,7 +142,13 @@ def _get_search_request(params: SearchOpportunityParams) -> dict:
 
     # Query
     if params.query:
-        builder.simple_query(params.query, SEARCH_FIELDS)
+        match params.experimental.scoring_rule:
+            case ScoringRuleEnum.EXPANDED:
+                builder.simple_query(params.query, EXPANDED)
+            case ScoringRuleEnum.AGENCY:
+                builder.simple_query(params.query, AGENCY)
+            case ScoringRuleEnum.DEFAULT:
+                builder.simple_query(params.query, DEFAULT)
```

Comment:
Hmm, something about this is bothering me. I'm not sure I like having effectively the same function call with one param difference setup like this.

My first thought was something like:
```py
        match params.experimental.scoring_rule:
            case ScoringRuleEnum.EXPANDED:
                filter_rule = EXPANDED
            case ScoringRuleEnum.AGENCY:
                filter_rule = AGENCY
            case ScoringRuleEnum.DEFAULT:
                filter_rule = DEFAULT

         builder.simple_query(params.query, filter_rule)
```

But then this is just a more verbose version of:

```py
# defined somewhere up top
FILTER_RULE_MAPPING = {
           ScoringRuleEnum.EXPANDED: EXPANDED,
           ScoringRuleEnum.AGENCY: AGENCY,
           ScoringRuleEnum.DEFAULT: DEFAULT
}

...

filter_rule = FILTER_RULE_MAPPING.get(params.experimental.scoring_rule, ScoringRuleEnum.DEFAULT)
builder.simple_query(params.query, filter_rule)
```

Thoughts on this approach?